magenta-lib/src/main/scala/magenta/deployment_type/GCS.scala

```diff
@@ -125,16 +167,43 @@ object GCS extends DeploymentType {
         stage = if (prefixStage(pkg, target, reporter)) Some(target.parameters.stage) else None,
         packageName = if (prefixPackage(pkg, target, reporter)) Some(pkg.name) else None
       ))
-      List(
-        GCSUpload(
-          bucket = bucketName,
-          paths = Seq(pkg.s3Package -> prefix),
-          cacheControlPatterns = cacheControl(pkg, target, reporter),
-          publicReadAcl = publicReadAcl(pkg, target, reporter),
-        )
+
+
+    val bucketConfig =  GcsTargetBucket( bucketName, target.parameters.stage.name, prefix,
+      directoriesToPruneByStage.get(pkg), fileTypesToPruneByStage.get(pkg))
+
+    List(
+      GCSUpload(
+        gcsTargetBucket = bucketConfig,
+        paths = Seq(pkg.s3Package -> prefix),
+        cacheControlPatterns = cacheControl(pkg, target, reporter),
+        publicReadAcl = publicReadAcl(pkg, target, reporter),
       )
-    }
+    )}
   }
 
   def defaultActions = List(uploadStaticFiles)
-}
\ No newline at end of file
+}
+
+object GcsTargetBucket {
+
+  def apply(name: String,
+            stage: String,
+            prefix: String,
+            maybeDirectories: Option[Map[String, List[String]]],
+            maybeFileTypes: Option[Map[String, List[String]]]): GcsTargetBucket = {
+
+      def listOrEmpty(maybeMap: Option[Map[String, List[String]]]) =
+        (for{
+          map <- maybeMap
+          value <- map.get(stage)
+        } yield value).getOrElse(List.empty)
```

Comment:
Should this be private for now?

Reply:
Yup good call!

Reply:
No actually - because it's enclosed in the `apply` method it's already private to there